boards/arm/xiao_ble/doc/index.rst

```diff
@@ -82,11 +82,29 @@ LED
 Programming and Debugging
 *************************
 
-The XIAO BLE ships with a bootloader. However, this guide doesn't describe how
-to use it and instead an External Debug Probe is used in order to program the
-board. If you have one, you can also use an external :ref:`debug probe
-<debug-probes>` to flash and debug Zephyr applications, but you need to solder
-an SWD header onto the back side of the board.
+The XIAO BLE ships with the `Adafruit nRF52 Bootloader`_ which supports flashing
+using `UF2`_. Doing so allows easy flashing of new images, but does not support
+debugging the device.
+
+UF2 Flashing
+============
+
+To enter the bootloader, connect the USB port of the XIAOB BLE to your host, and
+double tap the reset botton to the left of the USB connector. A mass storage
+device named `XIAO BLE` should appear on the host. Using the command line, or
+your file manager copy the `zephyr/zephyr.uf2` file from your build to the base
+of the `XIAO BLE` mass storage device. The XIAO BLE will automatically reset
+and launch the newly flashed application.
```

Comment:
Great addition! Tough
```
Flashing
========
```
section now needs to be either nested under

```
External Debugger
=================
```
or rephrased or something. It is misleading to have `UF2 Flashing` and `Flashing` at the same level, as it is now.

Reply:
Fixed, I changed both `Flashing` and `Debugging` to be nested one level deeper. That work for you?

Reply:
Perfect!